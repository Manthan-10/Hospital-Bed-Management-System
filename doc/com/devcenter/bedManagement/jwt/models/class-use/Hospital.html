<!DOCTYPE HTML>
<html lang="en">
<head>
<!-- Generated by javadoc (18) on Mon May 30 12:53:08 IST 2022 -->
<title>Uses of Class com.devcenter.bedManagement.jwt.models.Hospital</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2022-05-30">
<meta name="description" content="use: package: com.devcenter.bedManagement.jwt.models, class: Hospital">
<meta name="generator" content="javadoc/ClassUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-use-page">
<script type="text/javascript">var pathtoroot = "../../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon"></span><span class="nav-bar-toggle-icon"></span><span class="nav-bar-toggle-icon"></span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Class</a></li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Class com.devcenter.bedManagement.jwt.models.Hospital" class="title">Uses of Class<br>com.devcenter.bedManagement.jwt.models.Hospital</h1>
</div>
<div class="caption"><span>Packages that use <a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#com.devcenter.bedManagement.jwt.controllers">com.devcenter.bedManagement.jwt.controllers</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#com.devcenter.bedManagement.jwt.models">com.devcenter.bedManagement.jwt.models</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#com.devcenter.bedManagement.jwt.repository">com.devcenter.bedManagement.jwt.repository</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#com.devcenter.bedManagement.jwt.security.services">com.devcenter.bedManagement.jwt.security.services</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<section class="class-uses">
<ul class="block-list">
<li>
<section class="detail" id="com.devcenter.bedManagement.jwt.controllers">
<h2>Uses of <a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a> in <a href="../../controllers/package-summary.html">com.devcenter.bedManagement.jwt.controllers</a></h2>
<div class="caption"><span>Methods in <a href="../../controllers/package-summary.html">com.devcenter.bedManagement.jwt.controllers</a> with parameters of type <a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">HospitalController.</span><code><a href="../../controllers/HospitalController.html#createHospital(com.devcenter.bedManagement.jwt.models.Hospital)" class="member-name-link">createHospital</a><wbr>(<a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a>&nbsp;hospital)</code></div>
<div class="col-last even-row-color">
<div class="block">Function createHospital is mapped to /api/test/hospital/create for adding hospitals</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="com.devcenter.bedManagement.jwt.models">
<h2>Uses of <a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a> in <a href="../package-summary.html">com.devcenter.bedManagement.jwt.models</a></h2>
<div class="caption"><span>Methods in <a href="../package-summary.html">com.devcenter.bedManagement.jwt.models</a> that return <a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Floor.</span><code><a href="../Floor.html#getHospital()" class="member-name-link">getHospital</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Patient.</span><code><a href="../Patient.html#getHospital()" class="member-name-link">getHospital</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">com.devcenter.bedManagement.jwt.models</a> with parameters of type <a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Floor.</span><code><a href="../Floor.html#setHospital(com.devcenter.bedManagement.jwt.models.Hospital)" class="member-name-link">setHospital</a><wbr>(<a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a>&nbsp;hospital)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Patient.</span><code><a href="../Patient.html#setHospital(com.devcenter.bedManagement.jwt.models.Hospital)" class="member-name-link">setHospital</a><wbr>(<a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a>&nbsp;hospital)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../package-summary.html">com.devcenter.bedManagement.jwt.models</a> with parameters of type <a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../Floor.html#%3Cinit%3E(int,java.lang.String,com.devcenter.bedManagement.jwt.models.Hospital)" class="member-name-link">Floor</a><wbr>(int&nbsp;id,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;floorNo,
 <a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a>&nbsp;hospital)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../Patient.html#%3Cinit%3E(int,java.lang.String,java.lang.String,java.lang.String,java.util.Date,java.lang.String,java.lang.String,java.lang.String,com.devcenter.bedManagement.jwt.models.Hospital)" class="member-name-link">Patient</a><wbr>(int&nbsp;id,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;age,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;gender,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Date.html" title="class or interface in java.util" class="external-link">Date</a>&nbsp;dateOfBirth,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;address,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;phone,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;aadhaar,
 <a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a>&nbsp;hospital)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="com.devcenter.bedManagement.jwt.repository">
<h2>Uses of <a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a> in <a href="../../repository/package-summary.html">com.devcenter.bedManagement.jwt.repository</a></h2>
<div class="caption"><span>Methods in <a href="../../repository/package-summary.html">com.devcenter.bedManagement.jwt.repository</a> with parameters of type <a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FloorRepository.</span><code><a href="../../repository/FloorRepository.html#getFloorsByHospital(com.devcenter.bedManagement.jwt.models.Hospital)" class="member-name-link">getFloorsByHospital</a><wbr>(<a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a>&nbsp;hosp)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Patient.html" title="class in com.devcenter.bedManagement.jwt.models">Patient</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">PatientRepository.</span><code><a href="../../repository/PatientRepository.html#getPatientsByHospital(com.devcenter.bedManagement.jwt.models.Hospital)" class="member-name-link">getPatientsByHospital</a><wbr>(<a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a>&nbsp;hosp)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="com.devcenter.bedManagement.jwt.security.services">
<h2>Uses of <a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a> in <a href="../../security/services/package-summary.html">com.devcenter.bedManagement.jwt.security.services</a></h2>
<div class="caption"><span>Methods in <a href="../../security/services/package-summary.html">com.devcenter.bedManagement.jwt.security.services</a> that return <a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">HospitalService.</span><code><a href="../../security/services/HospitalService.html#createHospital(com.devcenter.bedManagement.jwt.models.Hospital)" class="member-name-link">createHospital</a><wbr>(<a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a>&nbsp;hospital)</code></div>
<div class="col-last even-row-color">
<div class="block">Function adds new hospitals</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">HospitalService.</span><code><a href="../../security/services/HospitalService.html#getHospitalById(int)" class="member-name-link">getHospitalById</a><wbr>(int&nbsp;id)</code></div>
<div class="col-last odd-row-color">
<div class="block">Function fetches the details of the hospital for the given id</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../security/services/package-summary.html">com.devcenter.bedManagement.jwt.security.services</a> that return types with arguments of type <a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">HospitalService.</span><code><a href="../../security/services/HospitalService.html#getHospitals()" class="member-name-link">getHospitals</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Function fetches all hospitals</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../security/services/package-summary.html">com.devcenter.bedManagement.jwt.security.services</a> with parameters of type <a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">HospitalService.</span><code><a href="../../security/services/HospitalService.html#createHospital(com.devcenter.bedManagement.jwt.models.Hospital)" class="member-name-link">createHospital</a><wbr>(<a href="../Hospital.html" title="class in com.devcenter.bedManagement.jwt.models">Hospital</a>&nbsp;hospital)</code></div>
<div class="col-last even-row-color">
<div class="block">Function adds new hospitals</div>
</div>
</div>
</section>
</li>
</ul>
</section>
</main>
</div>
</div>
</body>
</html>
