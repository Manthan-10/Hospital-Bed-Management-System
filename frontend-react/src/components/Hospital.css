.hospital{
    margin-top: 90px;
    margin-left: 270px;
    margin-right:10px;
    margin-bottom:0;
    background-color:white;
    padding: 20px;
   
   
   
}
.hospital-wrapper{
    display:flex;
    /* justify-content:center; */
    min-height:100px;
    float: right;
    

}

.hospital-heading{
    
   text-align: center;
   clear: right;
}
.body{
    background-color:grey;
    margin:0;
}