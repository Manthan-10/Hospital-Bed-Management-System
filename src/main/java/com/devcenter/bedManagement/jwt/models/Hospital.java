package com.devcenter.bedManagement.jwt.models;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

@Entity
@Table(name = "Hospital")
public class Hospital {
	

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	private String name;
	private String address;
	private String phone;
	
	@OneToMany(mappedBy="hospital", cascade = CascadeType.ALL)
	private List<Floor> floors;
	
	@OneToMany(mappedBy="hospital", cascade = CascadeType.ALL)
	private List<Patient> patients;
	
	public Hospital() {
		super();
	}

	public Hospital(int id, String name, String address, String phone) {
		super();
		this.id = id;
		this.name = name;
		this.address = address;
		this.phone = phone; 
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	} 

	public String getAddress() {
		return address;
	}

	public void setAddress(String address) {
		this.address = address;
	}

	public String getPhone() {
		return phone;
	}

	public void setPhone(String phone) {
		this.phone = phone;
	}

	
	

}
