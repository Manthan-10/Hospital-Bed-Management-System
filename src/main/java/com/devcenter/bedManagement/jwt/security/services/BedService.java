package com.devcenter.bedManagement.jwt.security.services;

import java.util.List;
import java.util.Optional;

import org.modelmapper.ModelMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.devcenter.bedManagement.jwt.models.Bed;
import com.devcenter.bedManagement.jwt.models.Room;
import com.devcenter.bedManagement.jwt.repository.BedRepository;
import com.devcenter.bedManagement.jwt.repository.RoomRepository;


/**
 * Service class
 * @author MN097789
 *
 */

@Service
public class BedService {
	
	@Autowired
	BedRepository bedRepository;
	
	@Autowired
	RoomRepository roomRepository;
	
	@Autowired
	private ModelMapper modelMapper;
	
	/**
	 * Function for adding new beds
	 * @param bed represents Bed object
	 * 	 
	 **/
	public void createBed(Bed bed) {
		 bedRepository.save(bed);
	}
	
	

	/**
	 * Function fetches all beds
	 * @return Returns List of all beds
	 */
	public List<Bed> getBeds() {
		return bedRepository.findAll(); 
	}
	
	
	/**
	 * Function fetches the details of all the beds for the given Room id 
	 * @param is id, which represents Room Id
	 * @return All Beds in that particular Room and its details
	 * 
	 */
	public List<Bed> getBedsByRoom(int id){
		
		Optional<Room> room = roomRepository.findById(id);
		List<Bed> beds = bedRepository.getBedsByRoom(this.modelMapper.map(room, Room.class));
		
		return beds;
		
	}
}
