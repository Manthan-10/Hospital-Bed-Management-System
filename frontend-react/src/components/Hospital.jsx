import React, { useEffect, useState } from "react";
import { useParams } from "react-router";
import FloorDropdown from "./FloorDropdown.jsx";
import HospitalDropdown from "./HospitalDropdown.jsx";
import "./Hospital.css";
import BedService from "../services/bed.service.js";
import { useTranslation } from "react-i18next";

/**
 * This component displays the dropdowns of both Hospital and Floor
 *
 */
const Hospital = () => {
  const { hid } = useParams();
  const [hospital, setHospital] = useState(null);
  const { t } = useTranslation();

  /**
   * @param {hid} HospitalId of that Hospital
   * @return Hospital Details of that hospital
   */
  useEffect(() => {
    BedService.getHospitalById(hid).then((response) => {
      console.log(response);
      setHospital(response);
    });
  }, []);

  return (
    <>
      <div className="hospital">
        <div className="hospital-wrapper">
          <div className="hospital-dropdown">
            <HospitalDropdown />
          </div>
          <div className="floor-dropdown">
            <FloorDropdown hid={hid} />
          </div>
        </div>

        {hospital && (
          <p className="hospital-heading">
            {hospital.name} {t("Hospital")}
          </p>
        )}
      </div>
    </>
  );
};

export default Hospital;
