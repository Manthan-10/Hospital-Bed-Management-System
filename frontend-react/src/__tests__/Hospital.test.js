import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Hospital from "../components/Hospital";
import Adapter from "enzyme-adapter-react-16";
import * as reactRouterDom from "react-router-dom";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { render } from "@testing-library/react";
import BedService from "../services/bed.service";

Enzyme.configure({ adapter: new Adapter() });

/**
 * Testing for UI of Hospital Component
 * @author MN097789
 */

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  useSelector: jest.fn(),

  useNavigate: () => mockNavigate,
}));
const mockNavigateSpy = jest.spyOn(reactRouterDom, "useNavigate");

const initialState = { output: 10 };
const mockStore = configureStore();
let store;

describe("Hospital UI Test Suite", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Hospital />);
  });

  test("Checks if it renders correctly", () => {
    store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <Hospital />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should check for divs", () => {
    wrapper = shallow(<Hospital />);
    expect(wrapper.length).toEqual(1);
  });

  test("BedService.getHospitalById is called", () => {
    jest.spyOn(React, "useEffect").mockImplementation((f) => f());
    jest.spyOn(BedService, "getHospitalById");
    jest.spyOn(global.console, "log");
    render(
      <Provider store={store}>
        <Hospital />
      </Provider>
    );

    expect(BedService.getHospitalById).toHaveBeenCalled();
  });
});
