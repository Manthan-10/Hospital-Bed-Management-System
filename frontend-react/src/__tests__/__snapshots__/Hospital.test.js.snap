// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hospital UI Test Suite Checks if it renders correctly 1`] = `
<div
  className="hospital"
>
  <div
    className="hospital-wrapper"
  >
    <div
      className="hospital-dropdown"
    >
      <div
        className="hsp-dropdown"
      >
        <select
          className="select-container"
          onChange={[Function]}
          size="large"
        >
          <option>
            Select_Hospital
          </option>
        </select>
      </div>
    </div>
    <div
      className="floor-dropdown"
    >
      <div
        className="floors"
      >
        <select
          className="select-container"
          onChange={[Function]}
          size="large"
        >
          <option>
            Select_Floor
             
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;
